tests/test_template.py

```diff
@@ -151,3 +154,63 @@ def test_substitute_with_dependencies():
                  'world': 'world'}
 
     assert Template.substitute_with_dependencies(inputs,inputs,TemplateConstants.DOLLAR_PARENTHESES)==output
+
+
+
+
+"""
+Unit tests for testing J2Template Class
+"""
+
+
+def test_dump_file():
+    """
+         Write rendered template to the output_path provided
+         Parameters
+         ----------
+         output_path : Path
```

Comment:
I don't think we need parameter lists in the doc blocks for the tests since they don't take any arguments. It may also be good to ensure that the doc block description is accurate based on the latest set of tests you have here. For example, this one tests that the template is rendered in memory from an input string template, and checks that the undeclared_variables returns the expected list.

I don't want to get ahead of myself too far, but this test could certainly be extended to test the dump method using a temporary file.